packages/babel-plugin/src/utils/css-builders.ts

```diff
@@ -619,6 +620,118 @@ const extractObjectExpression = (node: t.ObjectExpression, meta: Metadata): CSSO
   return { css: mergeSubsequentUnconditionalCssItems(css), variables };
 };
 
+/**
+ * Creates and returns `props` object
+ *
+ * @param deconstructedProp deconstructed prop
+ * @param parameters Group of parameters from deconstructed props
+ * @param prevKey to store keys of nested deconstructed props
+ */
+const getPropsNotDestructured = (
+  deconstructedProp: string,
+  parameters: FunctionParameters[],
+  prevKey?: string
+): string => {
+  for (const param of parameters) {
+    if (Array.isArray(param.value)) {
+      prevKey = param.key + '.';
+      getPropsNotDestructured(deconstructedProp, param.value, prevKey);
+    }
+
+    const parameter = parameters.find((x) => x.value == deconstructedProp);
+    if (parameter) {
+      return prevKey ? prevKey + parameter.key : parameter.key;
+    }
+  }
+
+  return prevKey ? prevKey + deconstructedProp : deconstructedProp;
+};
+/**
+ * Generates array of parameters from deconstructed `props`
+ *
+ * @param node ArrowFunction that has deconstructed `props` as parameters
+ * @returns Array of parameters
+ */
+const getParametersFromDestructuredProps = (node: t.ObjectPattern): FunctionParameters[] => {
+  let parameters: FunctionParameters[] = [];
+  const properties = node.properties;
+  parameters = properties
+    .map((po) => {
+      if (t.isObjectProperty(po)) {
+        const params: FunctionParameters = {
+          key: getKey(po.key),
+          value: t.isObjectPattern(po.value)
+            ? getParametersFromDestructuredProps(po.value)
+            : getKey(po.value as t.Expression),
+        };
+        return params;
+      }
+      return;
+    })
+    .filter(Boolean) as FunctionParameters[];
+  return parameters;
+};
+
+/**
+ * Convert `consequent` and `alternate` of Conditional expressions into Member Expression
+ * @param node Expression which has to be converted into Member Expression
+ * @param parameters Deconstructored props
+ */
+const modifyTemplateLiterals = (node: t.Expression, parameters: FunctionParameters[]) => {
+  if (t.isTemplateLiteral(node)) {
+    let count = -1;
```

Comment:
You can probably use `node.expressions[0]` directly instead of the loop.

Reply:
removed for loop